### PR TITLE
Support the mypy.

### PR DESCRIPTION
**Pull request by teweitsai**

### CODE REVIEW

File: python/lsst/ts/hexrotcomm/base_csc.py [hunk 1]
```diff
@@ -65,7 +73,7 @@ def make_connect_error_info(prefix, connected, connect_descr):
 
 # TODO DM-39787: remove this function once MTHexapod supports
 # MTRotator's simplified states.
-def make_state_transition_dict():
+def make_state_transition_dict() -> dict:
```
Comment (wvreeven) [inline python/lsst/ts/hexrotcomm/base_csc.py:76]: If possible, please specify the dict contents. Feel free to leave this for a future ticket.

File: python/lsst/ts/hexrotcomm/base_csc.py [hunk 2]
```diff
@@ -93,7 +101,7 @@ def make_state_transition_dict():
     }
 
     # compute transitions from non-FAULT to all other states
-    state_transition_dict = dict()
+    state_transition_dict: dict = dict()
```
Comment (wvreeven) [inline python/lsst/ts/hexrotcomm/base_csc.py:104]: If possible, please specify the dict contents. Feel free to leave this for a future ticket.

**Review by wvreeven — APPROVED** [review]
Looks good. I left two comments for you either for now or for a future ticket.

File: python/lsst/ts/hexrotcomm/base_csc.py (hunk 1 above)
Comment (teweitsai) [reply python/lsst/ts/hexrotcomm/base_csc.py]: Added in the commit: [4f2c704](https://github.com/lsst-ts/ts_hexrotcomm/pull/69/commits/4f2c704e3a773371b0ab61a92cfae6b9acd21c33). Thanks!

File: python/lsst/ts/hexrotcomm/base_csc.py (hunk 2 above)
Comment (teweitsai) [reply python/lsst/ts/hexrotcomm/base_csc.py:104]: This is the return value in the commit: [4f2c704](https://github.com/lsst-ts/ts_hexrotcomm/pull/69/commits/4f2c704e3a773371b0ab61a92cfae6b9acd21c33). Thanks!

**Pull request merged**